### PR TITLE
Remove wallet.sh

### LINKED ISSUE #1204: Get rid of multinode-demo/wallet.sh
The main reason `multinode-demo/wallet.sh` exists is to help with rsyncing leader.json.  But that shouldn't be necessary.  Instead we can use the same approach as `bench-tps` and give wallet a `--network` argument.

There's also a "reset" command that's implemented by `multinode-demo/wallet.sh`, which can just be deleted (with an update to `scripts/wallet-sanity.sh`).

Without `multinode-demo/wallet.sh`, the user will have to manually run `solana-keygen` and supply the `-k` argument, but that seems mostly fine.

### PR DESCRIPTION
**Pull request by CriesofCarrots**
Closes #1204 

Regarding keygen, right now if no keypair argument is passed, Wallet will look in `~/.config/solana` for a keypair file, and create it if it does not exist. I thought the `reset` argument was actually fairly useful, so I added it in to this default behavior.
@mvines , curious if you think this is viable.

### CODE REVIEW

File: src/bin/wallet.rs [hunk 1]
```diff
@@ -1,39 +1,50 @@
...
+        None
+    };
 
     let mut path = dirs::home_dir().expect("home directory");
     let id_path = if matches.is_present("keypair") {
         matches.value_of("keypair").unwrap()
     } else {
         path.extend(&[".config", "solana", "id.json"]);
+        if !path.exists() || matches.is_present("reset") {
+            Command::new("cargo")
```
Comment (mvines) [inline src/bin/wallet.rs]: Let's not assume `cargo` is available in the system.  Instead we can import the relevant solana-keygen code and invoke it here as a function.

File: multinode-demo/common.sh [hunk 2]
```diff
@@ -28,9 +28,9 @@ fi
 if [[ -d $SNAP ]]; then # Running inside a Linux Snap?
   solana_program() {
     declare program="$1"
-    if [[ "$program" = wallet || "$program" = bench-tps ]]; then
-      # TODO: Merge wallet.sh/client.sh functionality into
-      #       solana-wallet/solana-demo-client proper and remove this special case
+    if [[ "$program" = bench-tps ]]; then
+      # TODO: Merge client.sh functionality into
```
Comment (rob-solana) [inline multinode-demo/common.sh]: I think this comment can die, now.  The script is invoking bench-tps directly, not via client.sh.

Comment (CriesofCarrots) [reply multinode-demo/common.sh]: Am I correct that the whole `if` statement should die? Or just the comment?

Comment (mvines) [reply multinode-demo/common.sh]: 🍄 the whole `if` please

Comment (rob-solana) [reply multinode-demo/common.sh]: I'm not feeling so sure, now

Let me look at what command-%s.wrapper is...

File: src/bin/wallet.rs [hunk 3]
```diff
@@ -1,39 +1,44 @@
...
+        None
+    };
 
     let mut path = dirs::home_dir().expect("home directory");
     let id_path = if matches.is_present("keypair") {
         matches.value_of("keypair").unwrap()
     } else {
         path.extend(&[".config", "solana", "id.json"]);
+        if !path.exists() {
+            gen_keypair_file(path.to_str().unwrap().to_string())?;
```
Comment (mvines) [inline src/bin/wallet.rs]: Should we println a message to the user informing them of the new keyfile?

Comment (CriesofCarrots) [reply src/bin/wallet.rs]: Sure, I like that. Will add now.

File: multinode-demo/common.sh (hunk 2 above)
Comment (rob-solana) [reply multinode-demo/common.sh]: um, can't find it...  I think that means the clause should die?

**Pull request merged**